week-4/InClass/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,66 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 let destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"],
 };
 
 let destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"],
 };
 
 let destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"],
 };
 
 let destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"],
 };
 
-let travelDestinations = [destination1, destination2, destination3, destination4];
+let travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4,
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+let destinationNamesWithin500Kms = travelDestinations
+  .filter((place) => place.distanceKms <= 500)
+  .map((place) => place.destinationName); // Complete here
 
-let destinationNamesWithin500Kms = // Complete here
-
-let destinationNameReachableByFerry = // Complete here
-
-let destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
-
+let destinationNameReachableByFerry = travelDestinations
+  .filter((place) => place.transportations.includes("ferry"))
+  .map((place) => place.destinationName); // Complete here
 
+let destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(
+    (place) =>
+      place.distanceKms >= 300 && place.transportations.includes("train")
+  )
+  .map((place) => place.destinationName); // Complete here// Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
 /*
```

Comment:
You could put forEach on the next line, if you do it as part of the assignment it will return `undefined`,

Reply:
Thanks for your response. I have used the method forEach, but I am not sure that it is right way to solve this task.  Thanks for your assistance.

```
let destinationNamesMoreThan300KmsAwayByTrain = travelDestinations.filter((place) =>
      place.distanceKms >= 300 && place.transportations.includes("train")
  ).forEach((place) => console.log(place.destinationName)); 
```
Unfortunately, the variant with forEach does not pass the test and does not give expected result because it prints out city's names in a column.